oxalis-extension/oxalis-as2/src/main/java/network/oxalis/as2/outbound/As2MessageSender.java

```diff
@@ -228,45 +231,43 @@ protected HttpPost prepareHttpRequest() throws OxalisTransmissionException {
         } catch (MessagingException | IOException e) {
             throw new OxalisTransmissionException("Unexpected error during preparation of AS2 message.", e);
         } finally {
-            span.finish();
+            span.end();
         }
     }
 
     protected TransmissionResponse sendHttpRequest(HttpPost httpPost) throws OxalisTransmissionException {
-        Span span = tracer.buildSpan("execute").asChildOf(root).start();
+        Span span = tracer.spanBuilder("execute").startSpan();
         try (CloseableHttpClient httpClient = httpClientProvider.get()) {
             BasicHttpContext basicHttpContext = new BasicHttpContext();
-            basicHttpContext.setAttribute(Constants.PARENT_CONTEXT, span.context());
+            basicHttpContext.setAttribute("spanContext", span.getSpanContext());
```

Comment:
Better option is to replace below statement:
`basicHttpContext.setAttribute("spanContext", span.getSpanContext());`
with
`basicHttpContext.setAttribute(As2MessageSender.class.getName() + ".parentSpanContext", span.getSpanContext());`